cli/login.go

```diff
@@ -136,18 +136,26 @@ func (r *RootCmd) login() *clibase.Cmd {
 		useTokenForSession bool
 	)
 	cmd := &clibase.Cmd{
-		Use:        "login <url>",
+		Use:        "login [<url>]",
 		Short:      "Authenticate with Coder deployment",
 		Middleware: clibase.RequireRangeArgs(0, 1),
 		Handler: func(inv *clibase.Invocation) error {
 			ctx := inv.Context()
 			rawURL := ""
 			if len(inv.Args) == 0 {
 				rawURL = r.clientURL.String()
+				if rawURL != "" && rawURL == inv.Environ.Get(envURL) {
+					_, _ = fmt.Fprintf(inv.Stdout, "Attempting to authenticate with environment URL: %s\n", rawURL)
+				}
 			} else {
 				rawURL = inv.Args[0]
 			}
 
+			if url, err := r.createConfig().URL().Read(); rawURL == "" && err == nil {
```

Comment:
Should we also check that `url != ""` just in case? Otherwise we may print the statement below which is misleading.

Reply:
Potentially? I did it this way because if `url == ""` then that means the config/url exists but is empty so I think it makes more sense to print "Authing with config url: ". Maybe the url should be in quotes to indicate an empty url better.

I can see both sides here. Depends if you want to provide the extra info to the user or not.

Reply:
Actually. By changing when the output happens the case of the misleading print statement has been addressed.